juju/juju
# When removing consume permission from an offer, relations are set to suspended

## Description of change

When consume access is revoked on an offer for a user, any relations to that offer by that user are set to suspended. This is a new flag on the relation which will drive the uniter to act in a followup PR.

The suspend/resume CLI is also updated to set the new suspended flag.
To retain existing behaviour, the relation status is also set.

Relies on this description PR to land https://github.com/juju/description/pull/25

## QA steps

Deploy a cmr scenario, create an offer
Add a user, grant that user consume access to an offer
As that user, create a connection to the offer
As the admin, remove consume permission
Check that the relation state shows suspended in mongo
As the admin, try and resume the relation -> error because no consume permission
